chainguard-dev/edu
# glibc vs. musl

## Type of change

Adding a new article to discuss the differences between glibc and musl.

### What should this PR do?

This PR is apart of a Labs task to document differences between glibc and musl (related [issue](https://github.com/chainguard-dev/internal/issues/3358)):

1. Creates a new article discussing differences between glibc and musl.
2. Creates a new tab under Images ->  Working with Images -> Images for Compiled Programs -> glibc vs. musl. I have also moved the article "Images for Compiled Programs" under this subsection. 
3. Updates the "[Images for Compiled Programs](https://edu.chainguard.dev/chainguard/chainguard-images/working-with-images/compiled-programs)" to point users to this new glibc vs. musl for more details when discussing "[Why about musl?](https://edu.chainguard.dev/chainguard/chainguard-images/working-with-images/compiled-programs/#what-about-musl)" 

### Why are we making this change?

In my opinion, documentation surrounding glibc and musl has either been spread out across the internet and not well documented. Labs set out to create a concise document that highlights the differences between glibc and musl, and provide insights to users as to why Chainguard went with glibc over musl. Further insights can be found in this ticket: https://github.com/chainguard-dev/internal/issues/3358

### What are the acceptance criteria? 

Review from the deved team would be appreciated to confirm that I haven't messed up any headers (e.g., [here](https://github.com/tdunlap607/edu/blob/07c7104828e5bbfe7ca5ff2745e5ad0cb79f421b/content/chainguard/chainguard-images/working-with-images/compiled-programs/glibc-vs-musl/index.md?plain=1#L1)) or caused any breaking changes with the [new folder structure](https://github.com/tdunlap607/edu/tree/main/content/chainguard/chainguard-images/working-with-images/compiled-programs) under "Images for Compiled Programs." I tested with a local npm install and run, but would still appreciate a second set of eyes since this is my first PR into Chainguard Academy :) 

### How should this PR be tested?


- Please preview the article using the deploy preview link, click the links, make sure everything seems to connect where it should :)
- Any additional edits are much appreciated! I'm open to any suggestions on the text. Thanks!

cc @jspeed-meyers @sheesh
